privacyidea/lib/resolvers/LDAPIdResolver.py

```diff
@@ -399,7 +399,7 @@ def getUserList(self, searchDict):
                 comperator = ">="
                 if searchDict[search_key] in ["1", 1]:
                     comperator = "<="
-                filter += "(|({0!s}{1!s}{2!s})({3!s}!=0))".format(self.userinfo[search_key],
+                filter += "(&({0!s}{1!s}{2!s})(!({3!s}=0)))".format(self.userinfo[search_key],
                                                   comperator,
```

Comment:
Congratulations! You have me confused!
"accountExpires=1" is used by this script.
https://github.com/privacyidea/privacyidea/blob/master/tools/privacyidea-expired-users

It does not actually check, if the account is expired. At least in the script.
The script does it itself...
https://github.com/privacyidea/privacyidea/blob/master/tools/privacyidea-expired-users#L149

And then:
Why use (!({3!s}=0)) instead of ({3!s}!=0) ?